pkg/storage/snapshot/restore.go

```diff
@@ -495,7 +496,11 @@ func (ctrl *VMRestoreController) reconcileVolumeRestores(vmRestore *snapshotv1.V
 			if ownerDV != "" {
 				log.Log.Object(vmRestore).Infof("marking datavolume %s/%s as prepopulated before deleting its PVC", vmRestore.Namespace, ownerDV)
 
-				vmRestore.Status.Restores[i].DataVolumeName = &ownerDV
+				if vmRestore.Status.Restores[i].DataVolumeName == nil {
```

Comment:
I think its better here to do the update and mark waitingDVNameUpdate = true or something like that and then return that value at the end same as done for `waitingPVC` for example, so all the relevant restores will be updated in one time.
Keeping it like this will cause a new reconcile for each dv